File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/dto/hamburgesa/HamburgesaDto.java
package dto.hamburgesa;


import dto.ingrediente.IngredienteDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Root(name = "hamburgesa")
public class HamburgesaDto {
    @Attribute(name = "id")
    private String id;

    @Attribute(name = "nombre")
    private String nombre;

    @ElementList(name = "ingredientes", inline = true)
    private List<IngredienteDto> ingredientes;
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/dto/hamburgesa/HamburgesaDto.kt
package dto.hamburgesa

import dto.ingrediente.IngredienteDto
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.ElementList
import org.simpleframework.xml.Root

@Root(name = "hamburgesa")
data class HamburgesaDto (
    @field:Attribute(name = "id")
    @param:Attribute(name = "id")
    val id: String,

    @field:Attribute(name = "nombre")
    @param:Attribute(name = "nombre")
    val nombre: String,

    @field:ElementList(name = "ingredientes", inline = true)
    @param:ElementList(name = "ingredientes", inline = true)
    val ingredientes: List<IngredienteDto>
    )